mlx/backend/accelerate/softmax.cpp

```diff
@@ -274,7 +274,12 @@ void Softmax::eval_cpu(const std::vector<array>& inputs, array& out) {
 
   // Make sure that the last dimension is contiguous
   auto check_input = [](array x) {
-    if (x.strides()[x.ndim() - 1] == 1) {
+    bool no_copy = x.strides()[x.ndim() - 1] == 1;
+    if (x.ndim() > 1) {
+      auto s = x.strides()[x.ndim() - 1];
```

Comment:
I think this should be `x.ndim() - 2` btw.

Reply:
Ah shoot.. you're absolutely right

Reply:
Let me send another fix...